src/prefect/engine/serializers.py

```diff
@@ -48,7 +48,7 @@ def deserialize(self, value: Union[bytes, str]) -> Any:
                 raise exc
 
 
-class JSONSerializer(BytesSerializer):
+class JSONSerializer(PickleSerializer):
```

Comment:
Instead of having `JSONSerializer` subclass from `PickleSerializer`, we should define an empty base class `Serializer` defining the interface, and both implementations should be subclasses of that.